### PR TITLE
Add verification of `Quote3`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Adds logic to verify the contents of a `Quote3` quote with a provided
key. The `CertificationData` of the quote is not verified as it's from
this data that the quote signing key is retrieved.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #329** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/329" target="_blank">Graphite</a> 
    * **PR #330** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/330" target="_blank">Graphite</a> 
      * **PR #331** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/331" target="_blank">Graphite</a> 
        * **PR #333** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/333" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/333?utm_source=stack-comment).

File: dcap/types/src/quote3.rs [hunk 1]
```diff
@@ -215,7 +290,7 @@ impl<'a> TryFrom<&'a [u8]> for SignatureData<'a> {
 
         let (bytes, point_bytes) = take(KEY_SIZE)(bytes);
         let point = EncodedPoint::from_untagged_bytes(point_bytes.into());
-        let attestation_key = VerifyingKey::from_encoded_point(&point).unwrap();
+        let attestation_key = VerifyingKey::from_encoded_point(&point)?;
```
Comment (nick-mobilecoin) [inline dcap/types/src/quote3.rs:293]: This removal of an errant unwrap could be it's own PR, but it's fairly small and was noticed when creating tests for quote verification.

**Review by eranrund — APPROVED** [review]

**Review by cbeck88 — APPROVED** [review]
thank you
